In [1]:
import pandas as pd
df = pd.read_excel('MGTA456_BookDemand_Analysis1.xlsx')
print(df)


     Demand  Unnamed: 1 Summary Statistics   Unnamed: 3 Unnamed: 4
0     19283         NaN                NaN          NaN        NaN
1     18488         NaN            Average     19999.71        NaN
2     18653         NaN              Stdev  2000.195172        NaN
3     21183         NaN                Min        14986        NaN
4     21337         NaN                Max        25416        NaN
..      ...         ...                ...          ...        ...
495   20704         NaN                NaN          NaN        NaN
496   22824         NaN                NaN          NaN        NaN
497   22398         NaN                NaN          NaN        NaN
498   21751         NaN                NaN          NaN        NaN
499   17263         NaN                NaN          NaN        NaN

[500 rows x 5 columns]


In [3]:
import numpy as np
import pandas as pd

# Input parameters
retail_price = 22
printing_cost = 2
salvage_value = 0
optimal_order_quantity = 21824

# Calculate cost of understocking and overstocking
cost_of_understocking = retail_price - printing_cost
cost_of_overstocking = printing_cost - salvage_value

# Calculate target service level
target_service_level = cost_of_understocking / (cost_of_understocking + cost_of_overstocking)

# Read demand data from Excel file
demand_data = pd.read_excel('MGTA456_BookDemand_Analysis1.xlsx', usecols=[0], skiprows=1, nrows=500, names=['Demand'])

# Calculate profit for each demand scenario
def calculate_profit(demand):
    if demand < optimal_order_quantity:
        return (retail_price - printing_cost) * demand - (printing_cost * (optimal_order_quantity - demand))
    else:
        return (retail_price - printing_cost) * optimal_order_quantity

# Calculate profits for all demand scenarios
profits = demand_data['Demand'].apply(calculate_profit)

# Calculate average profit
average_profit = profits.mean()

# Print results
print(f"Cost of understocking (Cu): {cost_of_understocking}")
print(f"Cost of overstocking (Co): {cost_of_overstocking}")
print(f"Target service level: {target_service_level:.3f}")
print(f"Optimal order quantity (Q): {optimal_order_quantity}")
print(f"Average profit: {average_profit:.2f}")

Cost of understocking (Cu): 20
Cost of overstocking (Co): 2
Target service level: 0.909
Optimal order quantity (Q): 21824
Average profit: 392162.79


In [4]:
import numpy as np
import pandas as pd
from scipy.stats import norm

# Input parameters
retail_price = 22
printing_cost = 2
salvage_value = 0

# Calculate cost of understocking and overstocking
cost_of_understocking = retail_price - printing_cost
cost_of_overstocking = printing_cost - salvage_value

# Calculate target service level
target_service_level = cost_of_understocking / (cost_of_understocking + cost_of_overstocking)

# Read demand data from Excel file
demand_data = pd.read_excel('MGTA456_BookDemand_Analysis1.xlsx', usecols=[0], skiprows=1, nrows=500, names=['Demand'])

# Calculate mean and standard deviation of demand
mean_demand = demand_data['Demand'].mean()
std_demand = demand_data['Demand'].std()

# Calculate optimal order quantity using the newsvendor model
optimal_order_quantity = norm.ppf(target_service_level, loc=mean_demand, scale=std_demand)

# Calculate profit for each demand scenario
def calculate_profit(demand):
    if demand < optimal_order_quantity:
        return (retail_price - printing_cost) * demand - (printing_cost * (optimal_order_quantity - demand))
    else:
        return (retail_price - printing_cost) * optimal_order_quantity

# Calculate profits for all demand scenarios
profits = demand_data['Demand'].apply(calculate_profit)

# Calculate average profit
average_profit = profits.mean()

# Print results
print(f"Cost of understocking (Cu): {cost_of_understocking}")
print(f"Cost of overstocking (Co): {cost_of_overstocking}")
print(f"Target service level: {target_service_level:.3f}")
print(f"Optimal order quantity (Q): {optimal_order_quantity:.0f}")
print(f"Average profit: {average_profit:.2f}")

Cost of understocking (Cu): 20
Cost of overstocking (Co): 2
Target service level: 0.909
Optimal order quantity (Q): 22674
Average profit: 393133.28


As the manager of the entire supply chain, including both the publisher (Scribner) and the bookstore (BN), we need to consider the combined costs and profits across the entire chain.

The cost of understocking (Cu) in this context represents the opportunity cost or lost revenue when demand exceeds the order quantity. It includes the difference between the retail price and the wholesale price, as well as any associated costs incurred due to stockouts. Similarly, the cost of overstocking (Co) includes the costs associated with excess inventory, such as the difference between the wholesale price and the buy-back price.

To calculate the combined cost of understocking and overstocking for the supply chain, we need to consider the costs and profits for both the publisher and the bookstore. Additionally, we need to determine the target service level of the supply chain, which represents the desired probability of not running out of stock across both entities.

Based on the historical demand data and the newsvendor model, we can calculate the optimal order quantity for the supply chain. This quantity represents the number of books that should be printed to balance the risks of understocking and overstocking while maximizing the expected profit for the entire chain.

Once we have determined the optimal order quantity, we can calculate the average profit for the supply chain. This involves considering the revenue from sales at the retail price, the costs of printing and acquiring the books, and any buy-back revenues for unsold books.

By managing the entire supply chain, we can optimize the pricing strategy and inventory management practices to maximize the overall profitability of Scribner and BN. This integrated approach allows us to align the interests of both entities and efficiently meet customer demand while minimizing costs and maximizing profits across the entire chain.

In [5]:
import numpy as np
import pandas as pd
from scipy.stats import norm

# Read demand data from Excel file
demand_data = pd.read_excel('MGTA456_BookDemand_Analysis1.xlsx', usecols=[0], skiprows=1, nrows=500, names=['Demand'])

# Calculate mean and standard deviation of demand
mean_demand = demand_data['Demand'].mean()
std_demand = demand_data['Demand'].std()

# Input parameters
retail_price = 22
printing_cost = 2
current_wholesale_price = 14
current_buyback_price = 12

# Function to calculate the optimal order quantity for the bookstore
def calculate_optimal_order_quantity(wholesale_price, buyback_price):
    cost_of_understocking = retail_price - wholesale_price
    cost_of_overstocking = wholesale_price - buyback_price
    target_service_level = cost_of_understocking / (cost_of_understocking + cost_of_overstocking)
    return norm.ppf(target_service_level, loc=mean_demand, scale=std_demand)

# Function to calculate the expected profit for the bookstore
def calculate_bookstore_profit(wholesale_price, buyback_price, order_quantity):
    expected_sales = np.minimum(order_quantity, demand_data['Demand'])
    expected_profit = (retail_price - wholesale_price) * expected_sales.mean() + buyback_price * (order_quantity - expected_sales).mean()
    return expected_profit

# Function to calculate the expected profit for the publisher
def calculate_publisher_profit(wholesale_price, buyback_price, order_quantity):
    expected_sales = np.minimum(order_quantity, demand_data['Demand'])
    expected_profit = (wholesale_price - printing_cost) * order_quantity - buyback_price * (order_quantity - expected_sales).mean()
    return expected_profit

# Calculate the current expected profits for the bookstore and the publisher
current_order_quantity = calculate_optimal_order_quantity(current_wholesale_price, current_buyback_price)
current_bookstore_profit = calculate_bookstore_profit(current_wholesale_price, current_buyback_price, current_order_quantity)
current_publisher_profit = calculate_publisher_profit(current_wholesale_price, current_buyback_price, current_order_quantity)

# Search for a better price combination
best_wholesale_price = None
best_buyback_price = None
best_bookstore_profit = current_bookstore_profit
best_publisher_profit = current_publisher_profit

for wholesale_price in np.arange(13, 16, 0.1):
    for buyback_price in np.arange(11, 14, 0.1):
        order_quantity = calculate_optimal_order_quantity(wholesale_price, buyback_price)
        bookstore_profit = calculate_bookstore_profit(wholesale_price, buyback_price, order_quantity)
        publisher_profit = calculate_publisher_profit(wholesale_price, buyback_price, order_quantity)
        
        if bookstore_profit > best_bookstore_profit and publisher_profit > best_publisher_profit:
            best_wholesale_price = wholesale_price
            best_buyback_price = buyback_price
            best_bookstore_profit = bookstore_profit
            best_publisher_profit = publisher_profit

# Print the results
print(f"Current wholesale price: {current_wholesale_price:.2f}")
print(f"Current buy-back price: {current_buyback_price:.2f}")
print(f"Current bookstore profit: {current_bookstore_profit:.2f}")
print(f"Current publisher profit: {current_publisher_profit:.2f}")
print()
print(f"Best wholesale price: {best_wholesale_price:.2f}")
print(f"Best buy-back price: {best_buyback_price:.2f}")
print(f"Best bookstore profit: {best_bookstore_profit:.2f}")
print(f"Best publisher profit: {best_publisher_profit:.2f}")

Current wholesale price: 14.00
Current buy-back price: 12.00
Current bookstore profit: 181109.59
Current publisher profit: 237368.12

Best wholesale price: 14.60
Best buy-back price: 13.90
Best bookstore profit: 186361.16
Best publisher profit: 247567.31


In the context of the case study discussed in class, where the bookstore, BN, operates independently from the publisher, we aim to optimize the pricing strategy to benefit both parties. As the bookstore owner, decisions regarding the quantity of books to order are crucial, considering factors such as wholesale price and buy-back price. The buy-back contract, with a set buy-back price of $12, serves as a reference point for our analysis.

Utilizing historical demand data and leveraging the provided Python script, we sought to identify a pair of wholesale and buy-back prices that would improve the profitability of both the bookstore and the publisher compared to the current wholesale price of $14 and buy-back price of $12.

By iterating over various combinations of wholesale and buy-back prices, adjusting the wholesale price to $14.60 and the buy-back price to $13.90 would result in enhanced profitability for both parties. This optimization led to an increase in expected profits for both the bookstore and the publisher.

Insights:

Data-Driven Decision Making: The analysis underscores the importance of utilizing historical demand data to inform pricing decisions. By incorporating data-driven insights, we can identify opportunities to optimize pricing strategies and improve profitability.
Win-Win Scenario: The findings demonstrate that adjusting the pricing strategy can create a win-win scenario for both the bookstore and the publisher. By aligning incentives and optimizing pricing parameters, both parties can benefit from increased profitability.
Supply Chain Dynamics: The case study highlights the interconnected nature of supply chain dynamics between the bookstore and the publisher. Optimizing pricing strategies requires considering the implications for both parties and finding mutually beneficial solutions.
